packages/@lwc/synthetic-shadow/src/shared/retargetable.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+const retargetableEvents: Set<Event> = new Set();
+
+export function enableRetargeting(event: Event) {
+    retargetableEvents.add(event);
+}
+
+export function disableRetargeting(event: Event) {
+    retargetableEvents.delete(event);
+}
+
+export function isRetargetable(event: Event) {
+    return retargetableEvents.has(event);
```

Comment:
This should solve the asynchronous case. EventPhase is [supported in all browsers](https://developer.mozilla.org/en-US/docs/Web/API/Event/eventPhase#browser_compatibility).
```suggestion
    return retargetableEvents.has(event) || eventPhaseGetter.call(event) === Event.NONE;
```

Reply:
This doesn't work for listeners that expect retargeting that were added after our polyfill.

Reply:
Since it's impossible to know which listener an event reference came from, our options for async access are to either always retarget or never retarget. The former risks breaking handlers that are doing the right thing, while the latter risks breaking handlers that were added before LWC loads. Either way, it could be argued that async access is a corner case.

I prefer to always retarget for async access.